cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -349,6 +360,7 @@ void update(DatasetId instance, Map<String, String> properties) throws Exception
    *  have {@link Action#ADMIN} privileges on the #instance
    */
   void drop(DatasetId instance) throws Exception {
+    ensureAccess(instance);
```

Comment:
That will expose the existence of the dataset instance if user does not have any privilege on the dataset, which is not we want, right?